### PR TITLE
ayatana-indicator-sound: init & add to Lomiri

### PR DESCRIPTION
**Pull request by OPNA2608**
## Description of changes

An indicator for sound device control.

https://github.com/AyatanaIndicators/ayatana-indicator-sound

image

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by wegank** [comment]
@ofborg build ayatana-indicator-sound ayatana-indicator-sound.passthru.tests

**Comment by OPNA2608** [comment]
@ofborg build nixosTests.lomiri

**Comment by OPNA2608** [comment]
Okay, lomiri VM test is seemingly consistently failing on aarch64 ofborg when it tries to use OCR on indicator-client. That app behaves pretty inconsistently with only a keyboard so it kind of sucks for this kind of testing, but it *used* to work well enough in previous runs. Might be better to put some time into that ydotool module PR, so I can rewrite the indicator part of the test to not have to rely on indicator-client…

**Comment by OPNA2608** [comment]
@ofborg build ayatana-indicator-sound ayatana-indicator-sound.passthru.tests

**Comment by OPNA2608** [comment]
@ofborg build nixosTests.lomiri

File: nixos/modules/services/desktop-managers/lomiri.nix [hunk 1]
```diff
@@ -38,6 +38,7 @@ in {
       ]);
     };
 
+    hardware.pulseaudio.enable = lib.mkDefault true;
```
Comment (SuperSandro2000) [inline nixos/modules/services/desktop-managers/lomiri.nix:41]: You are sure we want to use legacy pulseaudio by default and pipewire?

File: nixos/modules/services/desktop-managers/lomiri.nix [hunk 2]
```diff
@@ -73,6 +74,8 @@ in {
         ayatana-indicator-datetime
         ayatana-indicator-messages
         ayatana-indicator-session
+      ] ++ lib.optionals (with config; hardware.pulseaudio.enable || services.pipewire.pulse.enable) [
```
Comment (SuperSandro2000) [inline nixos/modules/services/desktop-managers/lomiri.nix:77]:
```suggestion
      ] ++ lib.optionals (config.hardware.pulseaudio.enable || config.services.pipewire.pulse.enable) [
```
nit

File: pkgs/by-name/ay/ayatana-indicator-sound/package.nix [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  buildInputs = [
+    accountsservice
+    glib
+    gobject-introspection
+    libayatana-common
+    libgee
+    libnotify
+    libpulseaudio
+    libxml2
+    systemd
```
Comment (SuperSandro2000) [inline pkgs/by-name/ay/ayatana-indicator-sound/package.nix:73]: Maybe we could slim this down to libsystemd?

**Review by SuperSandro2000 — COMMENTED** [review]
Please rebase

File: nixos/modules/services/desktop-managers/lomiri.nix (hunk 1 above)
Comment (OPNA2608) [reply nixos/modules/services/desktop-managers/lomiri.nix:41]: [UBports is using PA still](https://docs.ubports.com/en/latest/porting/configure_test_fix/Sound.html), as does Debian with their Lomiri packaging. I haven't had any issues with it so far, so I see no urgency to push for PW right now.

File: pkgs/by-name/ay/ayatana-indicator-sound/package.nix (hunk 3 above)
Comment (OPNA2608) [reply pkgs/by-name/ay/ayatana-indicator-sound/package.nix:73]: A `libsystemd` doesn't seem to exist.

File: nixos/modules/services/desktop-managers/lomiri.nix (hunk 2 above)
Comment (OPNA2608) [reply nixos/modules/services/desktop-managers/lomiri.nix]: Applied.

**Comment by OPNA2608** [comment]
Rebased.

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**